rust/lance-io/src/object_store/gcs_wrapper.rs

```diff
@@ -24,7 +24,25 @@ use object_store::{
 use rand::Rng;
 use tokio::io::AsyncWrite;
 
-const MAX_UPLOAD_PARALLELISM: usize = 10;
+fn max_upload_parallelism() -> usize {
+    static MAX_UPLOAD_PARALLELISM: OnceLock<usize> = OnceLock::new();
+    *MAX_UPLOAD_PARALLELISM.get_or_init(|| {
+        std::env::var("LANCE_UPLOAD_CONCURRENCY")
+            .ok()
+            .and_then(|s| s.parse::<usize>().ok())
+            .unwrap_or(10)
+    })
+}
+
+fn max_conn_reset_retries() -> u16 {
+    static MAX_CONN_RESET_RETRIES: OnceLock<u16> = OnceLock::new();
+    *MAX_CONN_RESET_RETRIES.get_or_init(|| {
+        std::env::var("LANCE_CONN_RESET_RETRIES")
+            .ok()
+            .and_then(|s| s.parse::<u16>().ok())
+            .unwrap_or(20)
+    })
+}
```

Comment:
Cool, does this mean no more `lazy_static`?

Reply:
Yeah this is what we can do instead of lazy_static.